public/app/plugins/datasource/loki/datasource.ts

```diff
@@ -466,6 +466,10 @@ export class LokiDatasource
     return { start: timeRange.from.valueOf() * NS_IN_MS, end: timeRange.to.valueOf() * NS_IN_MS };
   }
 
+  isCompleteQuery(query?: LokiQuery | undefined): boolean {
```

Comment:
I understand this is a first pass solution, but I am wondering, if you would create an issue for us to improve it. We have parser for Loki - can see if query is valid/or not, so we could definitely improve it.

Reply:
Let's make sure we are on the same page with completeness vs validity first. Datasource maintainers can of course decide for themselves what this logic is, but we want this check to a.) prevent a query from being saved to query history b.) possibly prevent the query from being ran, not communicate the user typed in something wrong